view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js

```diff
@@ -247,6 +285,52 @@ define(
 
                 return false;
             },
+
+            processOrderWithCard: function (id) {
+                var self = this;
+
+                self.getPlaceOrderDeferredObject()
+                    .fail(
+                        function(response) {
+                            errorProcessor.process(response, self.messageContainer);
+                            fullScreenLoader.stopLoader();
+                            self.isPlaceOrderActionAllowed(true);
+                        }
+                    ).done(
+                        function(response) {
+                            if (self.isThreeDSecureEnabled()) {
+                                var serviceUrl = urlBuilder.createUrl(
+                                    '/orders/:order_id/omise-offsite',
+                                    {
+                                        order_id: response
+                                    }
+                                );
+
+                                storage.get(serviceUrl, false)
+                                    .fail(
+                                        function (response) {
+                                            errorProcessor.process(response, self.messageContainer);
+                                            fullScreenLoader.stopLoader();
+                                            self.isPlaceOrderActionAllowed(true);
+                                        }
+                                    )
+                                    .done(
+                                        function (response) {
+                                            if (response) {
+                                                $.mage.redirect(response.authorize_uri);
+                                            } else {
+                                                errorProcessor.process(response, self.messageContainer);
+                                                fullScreenLoader.stopLoader();
+                                                self.isPlaceOrderActionAllowed(true);
+                                            }
+                                        }
+                                    );
+                            } else if (self.redirectAfterPlaceOrder) {
+                                redirectOnSuccessAction.execute();
+                            }
+                        }
+                    );
+            }
```

Comment:
Nesting a bit deep here, and repeated identical anonymous functions. Could pull some out into a var/const/named function

Reply:
@jonrandy saw @jacekstan is working on refactoring these `method-rendered` at https://github.com/omise/omise-magento/tree/javascript-payment-renderer-refactoring so I may leave this one to not conflict with the change.

Reply:
ok... agread